src/lib.rs

```diff
@@ -12,36 +12,56 @@ use std::{
 };
 
 pub mod prelude {
+    //! Re-exports of the most commonly used types and functions.
     pub use super::{
         animated_pager, animated_pager_backward, animated_pager_forward,
         animated_pager_with_direction, TransitionStyle, TransitionType,
     };
 }
 
-/// See [`TransitionStyle::t_type`]
+/// Type of transition animation.
+///
+/// Specifies the direction of the animation.
+///
+/// Used in [`TransitionStyle::t_type`].
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Default)]
 pub enum TransitionType {
-    /// The animated ui will be moved horizontally
+    /// The animated UI will move horizontally.
     #[default]
     HorizontalMove,
-    /// The animated ui will be moved vertically
+    /// The animated UI will move vertically.
     VerticalMove,
 }
 
 impl TransitionType {
-    fn generate_tstransform(&self, amount: f32, origin: Vec2) -> TSTransform {
+    /// Generates the [`TSTransform`] based on the transition type and animation amount.
+    ///
+    /// This function is used internally by [`page_transition`] to apply the transformation to the UI.
+    ///
+    /// # Parameters
+    /// - `amount`: The amount of translation to apply. Positive values move in one direction, negative in the opposite.
+    /// - `origin`: Currently unused. Intended for future extensions where the origin of transformation might be customizable.
+    fn generate_tstransform(&self, amount: f32, _origin: Vec2) -> TSTransform {
         match self {
             Self::HorizontalMove => TSTransform::from_translation(Vec2::new(amount, 0.)),
             Self::VerticalMove => TSTransform::from_translation(Vec2::new(0., amount)),
         }
     }
 }
 
-// TODO: Document
-/// The transition animation used by [`animated_pager`].
+/// Applies a page transition animation to the given UI contents.
+///
+/// This is the core animation function used by [`animated_pager`] family of functions.
+/// It handles the visual transformation of the UI based on the animation `time` and [`TransitionStyle`].
 ///
 /// # Parameters
-///  - `time` - Should be between `0.0` and `1.0`
+/// - `ui`: The current [`Ui`] context.
+/// - `time`: The current animation time, ranging from `0.0` (start) to `1.0` (end).
+/// - `style`: The [`TransitionStyle`] defining the animation parameters.
+/// - `invert_direction`: If `true`, inverts the direction of the animation. Useful for backward transitions.
```

Comment:
```suggestion
/// - `invert_direction`: If `true`, the animation goes "backwards" (ie. if transition type is [`TransitionType::HorizontalMove`], this decides whether the animation goes left or right).
```

imo the concept of "inverting" would benefit from being explained a bit more here